src/lib.rs

```diff
@@ -88,6 +88,46 @@ impl PyFailure {
     }
 }
 
+#[pyclass]
+#[derive(Clone)]
+struct PyExponentialRetryConfig {
+    #[pyo3(get, set)]
+    initial_interval: Option<u64>,
+    #[pyo3(get, set)]
+    max_attempts: Option<u32>,
+    #[pyo3(get, set)]
+    max_duration: Option<u64>,
+}
+
+#[pymethods]
+impl PyExponentialRetryConfig {
+    #[pyo3(signature = (initial_interval=None, max_attempts=None, max_duration=None))]
+    #[new]
+    fn new(
+        initial_interval: Option<u64>,
+        max_attempts: Option<u32>,
+        max_duration: Option<u64>,
+    ) -> Self {
+        Self {
+            initial_interval,
+            max_attempts,
+            max_duration,
+        }
+    }
+}
+
+impl From<PyExponentialRetryConfig> for RetryPolicy {
+    fn from(value: PyExponentialRetryConfig) -> Self {
+        RetryPolicy::Exponential {
+            initial_interval: Duration::from_millis(value.initial_interval.unwrap_or(10)),
+            max_attempts: value.max_attempts,
+            max_duration: value.max_duration.map(Duration::from_millis),
+            factor: 2.0,
+            max_interval: None,
```

Comment:
Why not exposing these two as well?

Reply:
we'll need them for the e2e if you want the test to complete quickly :) https://github.com/restatedev/sdk-rust/blob/main/test-services/src/failing.rs#L122

Also I think in particular `max_interval` is important, otherwise the retry delay with exponent 2 quickly evolves into some too long delay!